xl_auth/commands.py

```diff
@@ -172,3 +172,254 @@ def urls(url, order):
 
     for row in rows:
         click.echo(str_template.format(*row[:column_length]))
+
+
+@click.command()
+@with_appcontext
+def import_data():
+    """Read data from Voyager dump and BibDB API to create DB entities.
+
+    Creates:
+        - collections
+        - user accounts for collection managers
+        - permissions between the two
+
+    """
+    import requests
+
+    from .collection.forms import RegisterForm as CollectionRegisterForm
+    from .collection.models import Collection
+    from .permission.models import Permission
+    from .user.forms import RegisterForm as UserRegisterForm
+    from .user.models import User
+
+    def _get_collection_details_from_bibdb(code):
+        raw_bibdb_api_data = json.loads(requests.get(
+            'https://bibdb.libris.kb.se/api/lib?level=brief&sigel={}'.format(code)).content)
+        if raw_bibdb_api_data['query']['operation'] == 'sigel {}'.format(code):
+            click.echo('.', nl=False)
+        else:
+            click.echo('-', nl=False)
+            raise AssertionError('Lookup failed for sigel %r' % code)
+
+        bibdb_api_data = None
+        for chunk in raw_bibdb_api_data['libraries']:
+            if chunk['sigel'] == code:
+                if bibdb_api_data is not None:
+                    raise AssertionError('Duplicate results for sigel %r' % code)
+                bibdb_api_data = chunk
+        if not bibdb_api_data:
+            raise AssertionError('Zero results for sigel %r' % code)
+
+        if bibdb_api_data['type'] in {'library', 'bibliography'}:
+            category = bibdb_api_data['type']
+        else:
+            category = 'uncategorized'
+        # assert bibdb_api_data['country_code'] == 'se'
+
+        if bibdb_api_data['dept']:
+            friendly_name = '%s, %s' % (bibdb_api_data['name'], bibdb_api_data['dept'])
+        else:
+            friendly_name = bibdb_api_data['name']
+
+        assert bibdb_api_data['alive'] in {True, False}
+
+        collection = {
+            'friendly_name': friendly_name,
+            'code': bibdb_api_data['sigel'],
+            'category': category,
+            'active': not bool(bibdb_api_data['sigel_new']),
+            'replaces': bibdb_api_data['sigel_old'],
+            'replaced_by': bibdb_api_data['sigel_new']
+        }
+
+        if bibdb_api_data['date_created']:
+            collection['created_at'] = \
+                dt.datetime.strptime(bibdb_api_data['date_created'], '%Y-%m-%dT%H:%M:%S')
+
+        return collection
+
+    def _get_voyager_data():
+        raw_voyager_sigels_and_locations = requests.get(
```

Comment:
Is `requests` necessary here? It feels like we should be able to read that file from disc instead. Or is this to keep our data access consistent?

Reply:
Not necessary, but I thought it was more straight-forward loading it from network (like we do with BibDB data). And it avoids ...
1. figuring out a good place to put this blob-like file of 4000 lines in the repo, maybe a "/resources" dir?
2. slippery-slope of checking in misc "good to have" stuff!

Reply:
Not necessary, but I thought it was more straight-forward loading it from network (like we do with BibDB data). And it avoids ...
1. figuring out a good place to put this blob-like file of 4000 lines in the repo, maybe a "/resources" dir?
2. slippery-slope of checking in misc "good to have" stuff!